lib/rules/sotd/process-document.js

```diff
@@ -30,29 +30,28 @@ exports.check = function (sr, done) {
     ,   doMeanwhile:  function() { deprecatedAllowed = true; }
     ,   doAfter:      function() { deprecatedAllowed = false; }
     });
-    if ($sotd && $sotd.length) {
+    if (sotd) {
         var found = false
         ,   regex = new RegExp(BOILERPLATE_PREFIX + '.+' + BOILERPLATE_SUFFIX)
         ;
-        $sotd.parent().find("p").each(function () {
-            var $p = sr.$(this);
-            if (sr.norm($p.text()) === boilerplate && $p.find("a").attr("href") === procUri) {
+        sotd.querySelectorAll("p").forEach(function (p) {
+            if (sr.norm(p.textContent) === boilerplate && p.querySelector("a").getAttribute("href") === procUri) {
                 if (found) sr.error(self, "multiple-times", { process: proc });
                 else {
                     found = true;
                 }
             }
-            else if (deprecatedBoilerplate === sr.norm($p.text()) || deprecatedUri === $p.find('a').attr('href')) {
+            else if (deprecatedBoilerplate === sr.norm(p.textContent) || deprecatedUri === p.querySelector('a').getAttribute('href')) {
```

Comment:
`find()` returns a new Cheerio object whether the selector matches some elements or not. However, `querySelector` will return `null` if no element can be found so you need to check if it returns `null` (everywhere where we used `find`) before getting the attribute.
At the moment, an exception is thrown if you try to check a document.